js-core/homeworks/homework-5/src/main.js

```diff
@@ -0,0 +1,159 @@
+/*NASK 1
+ 1. Переместите 0 в конец массива, остальные числа должны остаться
+ неизменными
+ // concat
+ example:
+ [1,false,2,0,3,null,0,4,0,25] => [1, false, 2, 3, null, 4, 25, 0, 0, 0]
+ [ 'a', 0, 0, 'b', null, 'c', 'd', 0, 1, false, 0, 1, 0, 3, [], 0, 1, 9, 0, 0, {}, 0, 0, 9 ] => ["a","b",null,"c","d",1,false,1,3,[],1,9,{},9,0,0,0,0,0,0,0,0,0,0]
+ [ 0, 1, null, 2, false, 1, 0 ] => [1,null,2,false,1,0,0]
+ */
+
+let arr1 = [1, false, 2, 0, 3, null, 0, 4, 0, 25];
+let arr2 = [
+  'a',
+  0,
+  0,
+  'b',
+  null,
+  'c',
+  'd',
+  0,
+  1,
+  false,
+  0,
+  1,
+  0,
+  3,
+  [],
+  0,
+  1,
+  9,
+  0,
+  0,
+  {},
+  0,
+  0,
+  9
+];
+ 
+function moveZeroToEnd(arr) {
+	let zeroArray = [];
+
+    while (arr.indexOf(0) != -1) {
+      let index = arr.indexOf(0);//will return an index
+      let element = arr[index];
+      arr.splice(index, 1);
+      zeroArray.push(element);
+    }
+    arr = arr.concat(zeroArray);
+    return arr;
```

Comment:
Is it work fine on the second array?

Reply:
In console.log I have 10 zeros at the end of the second array

Reply:
http://joxi.ru/L2169ebU867n4A
Like this